server/src/main/java/com/google/coffeehouse/common/Person.java

```diff
@@ -37,6 +47,41 @@ private Person(Builder builder) {
     this.userId = builder.idGenerator.generateId();
   }
 
+  /** Overloaded static factory, calls other fromMap with null IdentifierGenerator. */
+  public static Person fromMap(Map personInfo) {
+    return Person.fromMap(personInfo, null);
+  }
+
+  /** 
+   * Creates a {@link Person} object from a Map with the relevant parameters as keys.
+   * @param personInfo the Map that contains the information used to construct the Person. At a
+   *     minimum this includes an {@code "email"} key that is mapped to a String that describes
+   *     the email of the Person to be created, as well as a {@code "nickname"} key that is mapped
+   *     to a String that describes the name of the Person to be created. A {@code "pronouns"} key
+   *     that is mapped to a String describing the pronouns of the Person can optionally be added.
+   * @param idGen the {@link IdentifierGenerator} used when constructing the Person. If
+   *     null, the default generator will be used
+   * @return the created Person
+   * @throws IllegalArgumentException if no valid {@code "email"} key or 
+   *     {@code "nickname"} key is defined
+   */
+  public static Person fromMap(Map personInfo, IdentifierGenerator idGen) {
+    String email = (String) personInfo.getOrDefault(EMAIL_FIELD_NAME, null);
+    String nickname = (String) personInfo.getOrDefault(NICKNAME_FIELD_NAME, null);
+    if (email == null || nickname == null) {
+      throw new IllegalArgumentException(NO_VALID_NICKNAME_EMAIL_FROM_MAP);
+    }
+
+    Person.Builder personBuilder = Person.newBuilder(email, nickname);
+    if (personInfo.containsKey(PRONOUNS_FIELD_NAME)) {
+      personBuilder.setPronouns((String) personInfo.get(PRONOUNS_FIELD_NAME));
+    }
+    if (idGen != null) {
+      personBuilder.setIdGenerator(idGen);
+    }
+    return personBuilder.build();
+  }
```

Comment:
Again, want to call out that you're doing a great job utilizing the builder pattern, this is super clean.